.githooks/pre-commit

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env sh
+
+echo "(pre-commit) Cleaning compiled files..."
+npm run clean
+
+echo "(pre-commit) Building backend..."
+npm run build:backend
+
+linter_exit_code=0
+
+echo "(pre-commit) Running linters..."
+npm run lintfix
```

Comment:
Hmm I think the right behaviour here should be to `npm run lint` first, then `lintfix` if that fails, so erroneous changes don't get committed at all. (better yet if we can get lintfix to return non-0 code in one go if something is fixed)

On the same lines, we can stash local, unstaged changes before running the checks, then pop them back out once done, to ensure the checks are run on committed history. (consider how this should interact with `lintfix` too)

Reply:
> stash

(and the case there is nothing to stash)

Reply:
On my end lintfix does return non-zero code if something cannot be fixed. Is that alright?

But I do agree that we should stash the files first before we run the linters and even the tests when pushing.

Reply:
> On my end lintfix does return non-zero code if something cannot be fixed.

This is the case for me too, its alright since the fix needs to come from the developer. What about the case something is fixed? (imo this should fail, as `lintfix` does not automatically add/commit the fixes for you)

Reply:
I seee. I will just run npm run lint instead then.

Reply:
That's another option!

Reply:
We can put up the `lintfix` as another item for someone to work on -- not a priority but nice to have.